src/shared/tpm2-util.c

```diff
@@ -1676,44 +1676,53 @@ int tpm2_unseal(const char *device,
         if (r < 0)
                 goto finish;
 
-        r = tpm2_make_policy_session(
-                        c.esys_context,
-                        primary,
-                        hmac_session,
-                        TPM2_SE_POLICY,
-                        hash_pcr_mask,
-                        pcr_bank,
-                        pubkey, pubkey_size,
-                        pubkey_pcr_mask,
-                        signature,
-                        !!pin,
-                        &session,
-                        &policy_digest,
-                        /* ret_pcr_bank= */ NULL);
-        if (r < 0)
-                goto finish;
+        for (unsigned i = TPM2_PCRS_MAX;; i--) {
```

Comment:
hmm, using TPM2_PCRS_MAX as max loop count sounds weird. I don't see the relationship?

add a new macro?

Reply:
Yes, I didn't know how many times should we retry, so I thought about 1 per PCR (24) even though it's kind of random, but a new macro makes more sense. What number do you say we should use as max?

Reply:
We measure lots of stuff now, multiple things into various PCRs, hence if you manage to race against every single one of those the number you need to try might grow much larger than 24.

So I would just use something like 15 or so initially. or 30. pick something. doesn't really matter

i think in the long run, we might end up fine tuning this further anyway. i.e. start with something conservative. And if this collisions grow too frequent we might even have to add some form of sleeping between retries (exponential backoff or random or so), hence we should not have any illusions that we'll pick a perfect value right-away.